core/src/test/java/com/linecorp/armeria/server/AnnotatedServiceImplicitHeadTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+public class AnnotatedServiceImplicitHeadTest {
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.annotatedService(new Object() {
+                @Get("/test")
+                public HttpResponse hello() {
+                    return HttpResponse.of("Hello");
+                }
+            });
+        }
+    };
+
+    @Test
+    void getAnnotationMatchesHeadMethod() {
+        final BlockingWebClient webClient = server.blockingWebClient();
+        final AggregatedHttpResponse res = webClient.head("/test");
```

Comment:
Should we test HEAD with a non-existent path? An assertion error is raised without the fix I commented on above.
https://github.com/line/armeria/blob/85a2f93c04d81c3aaa527656cabcc69319cfd218/core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java#L167

Reply:
Yes, it seems we need to test for above case. I added it. Thank you!